client/state/themes/actions.js

```diff
@@ -418,6 +421,51 @@ export function clearActivated( siteId ) {
 }
 
 /**
+ * Triggers a network request to activate a specific wpcom theme on a given Jetpack site.
+ * First step of the process is the installation of the theme on Jetpack site.
+ * Second step is the actuall activation.
+ *
+ * @param  {Number|String} siteId    Site ID
+ * @param  {String}        themeId   Theme ID, this should be standard id without -wpcom suffix.
+ * @param  {String}        source    The source that is reuquesting theme activation, e.g. 'showcase'
+ * @param  {Boolean}       purchased Whether the theme has been purchased prior to activation
+ * @return {Function}                Action thunk
+ */
+export function activateWpcomThemeOnJetpack( siteId, themeId, source = 'unknown', purchased = false ) {
+	//Add -wpcom suffix. This suffix tells the endpoint that we want to
+	//install WordPress.com theme. Without the suffix endpoint would look
+	//for theme in .org
+	const wpcomThemeId = themeId + '-wpcom';
+	return dispatch => {
+		dispatch( {
+			type: THEME_ACTIVATE_ON_JETPACK_REQUEST,
+			wpcomThemeId,
+			siteId,
+		} );
+
+		return wpcom.undocumented().installThemeOnJetpack( siteId, wpcomThemeId )
+			.then( () => {
+				dispatch( activateTheme( wpcomThemeId, siteId, source, purchased ) );
+			} )
+			.then( () => {
+				dispatch( {
```

Comment:
Seems we need to dispatch `themeActivated` here, as in regular activation?
https://github.com/Automattic/wp-calypso/blob/4002c791311884688e9829bbbac196323f5c0ca6/client/state/themes/actions.js#L360